Sources/UIExtensions/Components/SwiftUI/UIKit Wrappers/PageView.swift

```diff
@@ -0,0 +1,238 @@
+    ///              return the View representing each page.
+    public init(activePageIndex: Binding<DataCollection.Index>,
+                items: DataCollection,
+                @ViewBuilder content: @escaping (DataCollection.Element) -> PageType) {
+        self.receivedItems = items
+        self.content = content
+        self._activePageIndex = activePageIndex
+    }
+
+    public func makeCoordinator() -> PageViewCoordinator {
+        // See PageViewCoordinator class and updateUIViewController for more information
+        // about the lifecycle of all of these parts (PageView, PageViewCoordinator, UIPageViewController).
+        return PageViewCoordinator(
+            parent: self,
+            items: receivedItems,
+            initialIndex: activePageIndex
+        )
+    }
+
+    public func makeUIViewController(context: Context) -> UIPageViewController {
+        let pageViewController = UIPageViewController(
+            transitionStyle: .scroll,
+            navigationOrientation: .horizontal)
+
+        pageViewController.dataSource = context.coordinator.pageDataSource
+        pageViewController.delegate = context.coordinator.pageDataSource
+
+        context.coordinator.items = receivedItems
+
+        // Configure the `UIPageViewController` initially once.
+        pageViewController.setViewControllers(
+            [context.coordinator.viewController(for: context.coordinator.items[safe: activePageIndex] ?? context.coordinator.items.last)],
+            direction: .forward,
+            animated: false
+        )
+
+        return pageViewController
+    }
+
+    public func updateUIViewController(_ pageViewController: UIPageViewController, context: Context) {
+        // This method is called when the `PageView` is recreated by SwiftUI.
+        // We are responsible here for updating the `UIPageViewController` and
+        // the `PageViewCoordinator` to reflect the new configuration.
+
+        // We only have to change the UIPageViewController if the collection has changed (even when it's not the presented item, because we need
+        // to refresh "page before" and "page after"), or the collection is the same but the selected page is different.
+        guard context.coordinator.items != receivedItems ||
+            context.coordinator.lastKnownSelectedIndex != activePageIndex else {
+            return
+        }
+
+        context.coordinator.items = receivedItems
+
+        let formerIndex = context.coordinator.lastKnownSelectedIndex ?? context.coordinator.items.endIndex
+
+        let element = context.coordinator.items[safe: activePageIndex] ?? context.coordinator.items.last
+
+        let direction: UIPageViewController.NavigationDirection = activePageIndex > formerIndex
+            ? .forward
+            : .reverse
+
+        let animated = activePageIndex != formerIndex || activePageIndex >= context.coordinator.items.endIndex
```

Comment:
Have you checked if it's possible to see if we are in an `withAnimation` block and use animated: true/false depending on that?